File: output/llama3.1-8b_IO_1/avatar/Java/Python/atcoder_ABC152_E.py
import math
from functools import reduce

def gcd(a: int, b: int) -> int:
    while b != 0:
        a, b = b, a % b
    return a

def lcm(a: int, b: int) -> int:
    return a * b // gcd(a, b)

def mod_div(a: int, b: int, mod: int) -> int:
    return (a % mod) * pow(b, -1, mod) % mod

def main():
    N = int(input())
    A = [int(x) for x in input().split()]
    lcm_val = 1
    for ai in A:
        a = abs(ai)
        lcm_val = lcm(lcm_val, a)
    mod = 10**9 + 7
    mod_lcm = lcm_val % mod
    ans = 0
    for ai in A:
        ans = (ans + mod_div(mod_lcm, abs(ai), mod)) % mod
    print(ans)

if __name__ == "__main__":
    main()

File: output/llama3.1-8b_IO_1/avatar/Python/Java/atcoder_ABC152_E.java
import java.util.Scanner;
import java.math.BigInteger;

public class atcoder_ABC152_E {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = scanner.nextInt();
        int[] A = new int[N];
        for (int i = 0; i < N; i++) {
            A[i] = scanner.nextInt();
        }
        BigInteger lcm = BigInteger.ONE;
        for (int a : A) {
            lcm = lcm.multiply(BigInteger.valueOf(a)).divide(gcd(lcm, BigInteger.valueOf(a)));
        }
        int sum = 0;
        for (int s : A) {
            sum += pow(s, MOD - 2, MOD);
        }
        System.out.println((lcm.multiply(BigInteger.valueOf(sum))).mod(BigInteger.valueOf(MOD)).longValue());
    }

    public static long gcd(long a, long b) {
        if (b == 0)
            return a;
        return gcd(b, a % b);
    }

    public static int pow(int s, int e, int mod) {
        int result = 1;
        while (e > 0) {
            if ((e & 1) != 0)
                result = (int) (((long) result * s) % mod);
            s = (int) (((long) s * s) % mod);
            e >>= 1;
        }
        return result;
    }

    private static final int MOD = 1000000007;
}
